src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -125,12 +125,13 @@ public HttpConnectionDispatcher(ConnectionManager manager, PipelineFactory facto
 
                 var resultTask = await Task.WhenAny(endpointTask, transportTask);
 
-                // Observe any exception
-                resultTask.GetAwaiter().GetResult();
-
                 if (resultTask == endpointTask)
                 {
                     // Notify the long polling transport to end
+                    if (endpointTask.IsFaulted)
```

Comment:
This looks good to me. 

You may also want to copy bits from the .Close Action. It is basically `async void` and it is being run as a timer action (i.e. on a ThreadPool thread). We await the endPointTask there and if the task is faulted it crashes the process (or leave it and I will do it in my PR)